lib/homework3.ex

```diff
@@ -1,18 +1,179 @@
 defmodule Homework3 do
-  @moduledoc """
-  Documentation for `Homework3`.
-  """
+  use Supervisor
 
-  @doc """
-  Hello world.
+  def start() do
+    Supervisor.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  def init(_) do
+    Supervisor.init(
+      [
+        {Morning, []}
+      ],
+      strategy: :one_for_one
+    )
+  end
+end
+
+defmodule Morning do
+  use Supervisor
+
+  def start_link(how) do
+    Supervisor.start_link(__MODULE__, how, name: __MODULE__)
+  end
+
+  def init(how) do
+    IO.inspect(how, label: "Woke up")
+    mode = get_mode(how)
+
+    processes = [
+      {Hygiene, mode},
+      {Breakfast, mode},
+      {Clothing, []}
+    ]
+
+    Supervisor.init(processes,
+      strategy: :one_for_one
+    )
+  end
+
+  defp get_mode(:early), do: :slow
+  defp get_mode(:late), do: :fast
+  defp get_mode(_), do: :normal
+end
+
+defmodule Hygiene do
+  @default_counter 50
+  @duration 6000
+  @fee 5
+  use Task
+
+  defmodule Exception do
+    defexception message: "Imaginary parent: Not so fast, my child, #{@fee} more swipes."
+  end
+
+  def start_link(:slow) do
+    Task.start_link(fn -> brush_teeth(60) end)
+  end
+
+  def start_link(:fast) do
+    Task.start_link(fn -> brush_teeth(40) end)
+  end
+
+  def start_link(_) do
+    Task.start_link(fn -> brush_teeth(@default_counter) end)
+  end
+
+  defp clean_a_teeth() do
+    swipe_time = 50 + :rand.uniform(101 - 50)
+    Process.sleep(swipe_time)
+    swipe_time
+  end
+
+  def brush_teeth(swipe_counter, total_time \\ 0)
+
+  def brush_teeth(swipe_counter, total_time) when swipe_counter > 0 do
+    IO.puts("Me: I swiped the brush so many times, and #{swipe_counter} times to go!")
+    brush_teeth(swipe_counter - 1, total_time + clean_a_teeth())
+  end
+
+  def brush_teeth(swipe_counter, total_time) when swipe_counter === 0 do
+    swipe_time = 50 + :rand.uniform(101 - 50)
+    Process.sleep(swipe_time)
+    IO.puts("Me: Now my teeth are clean.")
+
+    try do
+      validate(total_time, swipe_counter)
+    rescue
+      Hygiene.Exception -> bargain(total_time)
+      _ -> IO.puts("Me: Whoopsy!")
+    end
+  end
+
+  defp validate(total_time, swipe_counter)
+       when total_time < @duration and swipe_counter == 0 do
```

Comment:
```suggestion
  defp validate(total_time, 0) when total_time < @duration do
```
Above you used `===/2` for the very same comparison, I understand JS hurts, but at least you should use either of them, not both simultaneously.